torchvision/utils.py

```diff
@@ -199,10 +199,16 @@ def draw_bounding_boxes(
 
     txt_font = ImageFont.load_default() if font is None else ImageFont.truetype(font=font, size=font_size)
 
+    if colors is None:
+        if labels is None:
+            colors = _generate_color_palette(len(img_boxes))
+        else:
+            assert len(labels) == len(img_boxes)
```

Comment:
But why do we want len(labels) to be same as len(boxes)?

I thought we should assume len(labels) < len(boxes).
As we do the same in drawing masks

```
if colors is not None and num_masks > len(colors):
    raise ValueError(f"There are more masks ({num_masks}) than colors ({len(colors)})")
```

I thought if a box is unlabelled, we will just draw a random color.

In case a box is not labelled 
```
if ! label in label_color_map.keys():
   colors.append(None) (Or generate a random color)
```

Reply:
> Is it really necessary for every box to be labelled?

I would say so. Another way to phrase the question is: Why would someone label the first N boxes, but not the rest?

I can't think of a good reason to do so, and I don't think this is something we should be bothered to support anyway. Users can always pass empty strings as the labels if they reaaaaally have a use-case where only the first few boxes have labels.

Reply:
I agree with Nicolas on this

Reply:
I suggest to replace plain assert by raising ValueError and suggesting user what Nicolas said